test/inductor/test_fp8.py

```diff
@@ -249,23 +268,29 @@ def amax_fp8(x: Tensor, scale: Tensor):
         compiled_amax_fp8_quant = torch.compile(amax_fp8, backend="inductor")
 
         x_shape = (batch_size, sequence_length, hidden_size)
-        x = torch.rand(*x_shape, device="cuda", dtype=torch.half)
-        scale = torch.tensor(0.2, device="cuda", dtype=torch.float)
+        x = torch.rand(*x_shape, device=device, dtype=torch.half)
+        scale = torch.tensor(0.2, device=device, dtype=torch.float)
 
         y_compiled = compiled_amax_fp8_quant(x, scale)
         y = amax_fp8(x, scale)
 
         torch.testing.assert_close(y_compiled.half(), y.half(), rtol=1e-2, atol=1e-2)
 
-    @unittest.skipIf(not PLATFORM_SUPPORTS_FP8, f8_msg)
     @parametrize(
         "float8_dtype",
         (torch.float8_e4m3fn, torch.float8_e5m2)
         if torch.version.hip is None
         else (torch.float8_e4m3fnuz, torch.float8_e5m2fnuz),
     )
     @parametrize("shape", ("1,1,15", "1,10,15", "1,10,512", "1,10,4096", "4,2048,4096"))
-    def test_amax_along_with_fp8_quant(self, float8_dtype: torch.dtype, shape: str):
+    @parametrize("device", ("cuda", "cpu"))
+    def test_amax_along_with_fp8_quant(
+        self, float8_dtype: torch.dtype, shape: str, device: torch.device
+    ):
+        if device == "cuda" and not PLATFORM_SUPPORTS_FP8:
+            raise unittest.SkipTest(f8_msg)
+        if device == "cpu" and TEST_WITH_ROCM:
+            raise unittest.SkipTest("Skip fp8 CPU tests in ROCm")
```

Comment:
This feels very wrong to me (also, run test with keep going and I suspect you'll see more failures on those devices, because fp8 includes are guarded incorrectly there)